src/components/tl/mlx5/mcast/tl_mlx5_mcast.h

```diff
@@ -183,65 +200,95 @@ struct packet {
+    struct ibv_recv_wr                         *call_rwr;
+    struct ibv_sge                             *call_rsgs;
+    uint64_t                                    timer;
+    int                                         stalled;
+    int                                         comm_id;
+    void                                       *p2p_ctx;
+    ucc_base_lib_t                             *lib;
+    struct sockaddr_in6                         mcast_addr;
+    ucc_rank_t                                  parents[MAX_COMM_POW2];
+    ucc_rank_t                                  children[MAX_COMM_POW2];
+    int                                         nack_requests;
+    int                                         nacks_counter;
+    int                                         n_prep_reliable;
+    int                                         n_mcast_reliable;
+    int                                         wsize;
+    ucc_tl_mlx5_mcast_join_info_t              *group_setup_info;
+    ucc_service_coll_req_t                     *group_setup_info_req;
+    ucc_status_t                               (*bcast_post) (void*, void*, size_t, ucc_rank_t, ucc_service_coll_req_t**);
+    ucc_status_t                               (*allgather_post) (void*, void*, void*, size_t, ucc_service_coll_req_t**);
+    ucc_status_t                               (*barrier_post) (void*, ucc_service_coll_req_t**);
+    ucc_status_t                               (*coll_test) (ucc_service_coll_req_t*);
+    ucc_service_coll_req_t                     *one_sided_reliability_req;
+    struct rdma_cm_event                       *event;
+    ucc_tl_mlx5_one_sided_reliable_team_info_t *one_sided_reliability_info;
+    ucc_tl_mlx5_mcast_slot_mem_info_t          *slot_mem_info_list;
+    uint32_t                                   *recv_list; //counter for each target recv packet
+    uint32_t                                   *remote_slot_info; // remote ag_counter to check if remote slots is
+                                                                  // ready for RDMA READ
+    struct ibv_mr                              *remote_slot_info_mr;
+    int                                         one_sided_reliability_enabled;
+    int                                         one_sided_reliability_ready;
+    int                                         one_sided_reliability_scheme_msg_threshold;
+    int                                         mcast_trigger_reliability;
+    struct ibv_mr                              *one_sided_slots_mr;
+    ucc_tl_mlx5_mcast_slot_mem_info_t          *one_sided_async_slots_info_list;
+    int                                         one_sided_async_slots_size;
+    char                                       *one_sided_slots_buffer;
+    int                                         one_sided_slots_state;
+    struct pp_packet                           *r_window[1]; // note: do not add any new variable after here
```

Comment:
Can we organize this a bit - this has got to be the biggest datastruct I ever saw haha.

Reply:
+1
Can you please indicate what fields are added in this pr?

Reply:
sure, I will create a new struct and move the new variables under that. 

here are the new vars that are being added:
```
    ucc_tl_mlx5_one_sided_reliable_team_info_t *one_sided_reliability_info;
    ucc_tl_mlx5_mcast_slot_mem_info_t          *slot_mem_info_list;
    uint32_t                                   *recv_list; //counter for each target recv packet
    uint32_t                                   *remote_slot_info; // remote ag_counter to check if remote slots is
                                                                  // ready for RDMA READ
    struct ibv_mr                              *remote_slot_info_mr;
    int                                         one_sided_reliability_enabled;
    int                                         one_sided_reliability_ready;
    int                                         one_sided_reliability_scheme_msg_threshold;
    int                                         mcast_trigger_reliability;
    struct ibv_mr                              *one_sided_slots_mr;
    ucc_tl_mlx5_mcast_slot_mem_info_t          *one_sided_async_slots_info_list;
    int                                         one_sided_async_slots_size;
    char                                       *one_sided_slots_buffer;
    int                                         one_sided_slots_state;
```

Reply:
i made the requested changes